test/unit/objectLiteral.spec.ts

```diff
@@ -63,3 +63,36 @@ test.each(['{x: "foobar"}.x', '{x: "foobar"}["x"]', '{x: () => "foobar"}.x()', '
         util.testExpression(expression).expectToMatchJsResult();
     }
 );
+
+test("noSelf in object literal functions", () => {
```

Comment:
I would also like to see a third test with the code from your issue: 

https://typescripttolua.github.io/play/#code/FDCWDsBcFMCcDMCGBjaACAYgV3MyoB7cAYSMkQjjQG9g1614CCAuNACkgAtQBnFgG4FQAEwA0aRG16RYEAOYBKALwA+IaLoMARolhtOPfhvGTpshSvXCRW+roBeB7n0E2JUtDLnglak8AAviDIRDJoALYAnti4pFAU4HBssXiEJGSJVMo0dozM7IiKtAylaKHgvAQANtAAdNUE8oWKeYFiebr6jDhpRC0lZfQVVbUNTS1tHaWOBkVqg0MjNfWNzUVtQSDRqfHklLB1TATsAEQ8p607vXtZh11nF1cxN5kHdY6PoJdAA